src/main/java/com/conveyal/datatools/manager/gtfsplus/GtfsPlusValidation.java

```diff
@@ -16,39 +16,60 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.InputStreamReader;
+import java.io.Serializable;
 import java.util.Arrays;
 import java.util.Collection;
+import java.util.Date;
 import java.util.Enumeration;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.zip.ZipEntry;
 import java.util.zip.ZipFile;
 
-public class GtfsPlusValidation {
-    public static final Logger LOG = LoggerFactory.getLogger(GtfsPlusValidation.class);
+/** Generates a GTFS+ validation report for a file. */
+public class GtfsPlusValidation implements Serializable {
+    private static final long serialVersionUID = 1L;
+    private static final Logger LOG = LoggerFactory.getLogger(GtfsPlusValidation.class);
     private static final FeedStore gtfsPlusStore = new FeedStore(DataManager.GTFS_PLUS_SUBDIR);
     private static final String NOT_FOUND = "not found in GTFS";
 
+    // Public fields to appear in validation JSON.
+    public final String feedVersionId;
+    /** Indicates whether GTFS+ validation applies to user-edited feed or original published GTFS feed */
+    public boolean published;
+    public long lastModified;
+    /** Issues found for this GTFS+ feed */
+    public List<ValidationIssue> issues = new LinkedList<>();
+
+    private GtfsPlusValidation (String feedVersionId) {
+        this.feedVersionId = feedVersionId;
+    }
+
     /**
      * Validate a GTFS+ feed and return a list of issues encountered.
      * FIXME: For now this uses the MapDB-backed GTFSFeed class. Which actually suggests that this might
      *   should be contained within a MonitorableJob.
      */
-    public static List<ValidationIssue> validateGtfsPlus (String feedVersionId) throws IOException {
+    public static GtfsPlusValidation validate(String feedVersionId) throws IOException {
+        GtfsPlusValidation validation = new GtfsPlusValidation(feedVersionId);
         if (!DataManager.isModuleEnabled("gtfsplus")) {
             throw new IllegalStateException("GTFS+ module must be enabled in server.yml to run GTFS+ validation.");
         }
-        List<ValidationIssue> issues = new LinkedList<>();
         LOG.info("Validating GTFS+ for " + feedVersionId);
+
         FeedVersion feedVersion = Persistence.feedVersions.getById(feedVersionId);
         // Load the main GTFS file.
         // FIXME: Swap MapDB-backed GTFSFeed for use of SQL data?
         GTFSFeed gtfsFeed = GTFSFeed.fromFile(feedVersion.retrieveGtfsFile().getAbsolutePath());
```

Comment:
This line is causing a crash when GTFS Plus info is attempted to be fetched from a version that was created from a snapshot that was immediately saved after creating the snapshot from the editor where the editor was started from scratch.